### PR TITLE
Parallelizing Resize op

### PR DESCRIPTION
**Pull request by duli2012**
**Description**: Describe your changes.
Make bilinear and trilinear mode of Resize parallel

**Motivation and Context**
- Why is this change required? What problem does it solve?
This change makes the resize op 3 times faster for a product model.

Credit to @tlh20

### CODE REVIEW

**Comment by duli2012** [comment]
Currently, it's using ThreadPool::TrySimpleParallelFor API. We'd like to use ThreadPool::TryParallelFor which is more robust and effcient. However,  the latter has unaddressed issues related Eigan cost model, which we'll revisit later.

File: onnxruntime/core/providers/cpu/tensor/upsample.cc [hunk 1]
```diff
@@ -408,39 +410,41 @@ void UpsampleBilinear(int64_t batch_size,
...
+                                                      }
+                                                    }
+                                                    Xdata += input_height * input_width;
+                                                    Ydata += output_width * output_height;
+                                                  });
   }
 }
 
-// The following method supports a 5-D input in 'Linear mode'
+  // The following method supports a 5-D input in 'Linear mode'
```
Comment (tlh20) [inline onnxruntime/core/providers/cpu/tensor/upsample.cc:447]: Nit : spurious space here

File: onnxruntime/core/providers/cpu/tensor/upsample.cc [hunk 2]
```diff
@@ -879,7 +887,7 @@ Status Upsample<T>::BaseCompute(OpKernelContext* context,
                                 scales, roi, is_resize_, use_extrapolation_, extrapolation_value_,
                                 use_nearest2x_optimization_, get_original_coordinate_, get_nearest_pixel_);
     case UpsampleMode::LINEAR: {
-      // Supports 'bilinear' and 'trilinear' sampling only
+      // Supports 'bilinear' and 'trilinear' sampling onl
```
Comment (tlh20) [inline onnxruntime/core/providers/cpu/tensor/upsample.cc:890]: Nit : onl->only

File: onnxruntime/core/providers/cpu/tensor/upsample.cc [hunk 3]
```diff
@@ -408,39 +410,41 @@ void UpsampleBilinear(int64_t batch_size,
...
-                                                       dx1[x] * dy1[y] * X22);
-        }
-      }
-      Xdata += input_height * input_width;
-      Ydata += output_width * output_height;
-    }
+  
+ for (int64_t n = 0; n < batch_size; ++n) {
+    concurrency::ThreadPool::TrySimpleParallelFor(tp, num_channels,
+                                                  [&](std::ptrdiff_t c) {
```
Comment (tlh20) [inline onnxruntime/core/providers/cpu/tensor/upsample.cc:416]: Only technical comment on the PR is whether we need to back off to a sequential execution when the output images are tiny?  Given that we are upsampling, is it OK to assume that we will not encounter that case?  One way to get sequential execution without having two branches of the code is to pass in tp=nullptr.  [ Ultimately we'd like this to be handled in the thread pool cost model, but per e-mail discussion, it looks like we need some investigation and tuning there .]

**Review by tlh20 — CHANGES_REQUESTED** [review]
Couple of minor formatting glitches.  Only substantive comment is if we need to fall back to sequential execution on any cases.

File: onnxruntime/core/providers/cpu/tensor/upsample.cc (hunk 2 above)
Comment (duli2012) [reply onnxruntime/core/providers/cpu/tensor/upsample.cc]: fixed

File: onnxruntime/core/providers/cpu/tensor/upsample.cc (hunk 3 above)
Comment (duli2012) [reply onnxruntime/core/providers/cpu/tensor/upsample.cc:416]: added a condition that it only parallelize if output size is greater than 64.

**Review by tlh20 — APPROVED** [review]

**Pull request merged**